src/dir.targets

```diff
@@ -157,9 +154,39 @@
     <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
   </PropertyGroup>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(IsTestProject)'=='True' And '$(SkipTests)'!='True'">
+    <PropertyGroup>
+      <!-- By default project has inner- and outer- loop tests. -->
+      <TestCategories Condition="'$(TestCategories)'==''">InnerLoop;OuterLoop</TestCategories>
+    </PropertyGroup>
+    <ItemGroup>
+      <!-- Split semicolon separated list -->
+      <TestCategoriesItems Include="$(TestCategories)" />
+      
+      <!-- If category not found then list will be empty -->
+      <InnerLoop Include="@(TestCategoriesItems)" Condition="'%(TestCategoriesItems.Identity)'=='InnerLoop'" />
+      <OuterLoop Include="@(TestCategoriesItems)" Condition="'%(TestCategoriesItems.Identity)'=='OuterLoop'" />
+    </ItemGroup>
+    <PropertyGroup>
+      <!-- If list not empty then category found -->
+      <IsInnerLoop Condition="'@(InnerLoop)'!=''">True</IsInnerLoop>
+      <IsOuterLoop Condition="'@(OuterLoop)'!=''">True</IsOuterLoop>
+
+      <!-- What categories of tests do we want to run by default  -->
+      <RunInnerLoop Condition="'$(RunInnerLoop)'==''">True</RunInnerLoop>
+      
+      <!-- Run tests only if we want to run tests of the given category and the test has this category -->
+      <RunTestsForProject>False</RunTestsForProject>
+      <RunTestsForProject Condition="'$(IsInnerLoop)'=='True' And '$(RunInnerLoop)'=='True'">True</RunTestsForProject>
+      <RunTestsForProject Condition="'$(IsOuterLoop)'=='True' And '$(RunOuterLoop)'=='True'">True</RunTestsForProject>
```

Comment:
You are going to need to move the RunOuterLoop properties into this target as well to make sure it gets included correctly. 

It would kind of be nice of these TestCategories got translated into traits (perhaps we even call the property TestTraits to make it align).